generate-historical-data.mjs

```diff
@@ -0,0 +1,416 @@
+  // Use the version date itself as the compatibility date
+  // This ensures the workerd binary supports this date
+  return `${year}-${month}-${day}`;
+}
+
+async function findMaxSupportedCompatibilityDate(workerdVersion) {
+  // Try the version's own date first
+  const versionDate = getCompatibilityDateForVersion(workerdVersion);
+  
+  try {
+    // Test if workerd supports this compatibility date
+    console.log(`Testing compatibility date: ${versionDate}`);
+    const testResult = shell.exec(`node workerd/dump.mjs ${versionDate}`, { silent: true });
+    console.log(`Test result code: ${testResult.code}`);
+    if (testResult.stderr) console.log(`Test stderr: ${testResult.stderr.substring(0, 200)}...`);
+    
+    if (testResult.code === 0) {
+      console.log(`✓ Using compatibility date: ${versionDate}`);
+      return versionDate;
+    }
+  } catch (error) {
+    console.log(`Error testing version date: ${error.message}`);
+  }
+  
+  // If the version date doesn't work, try progressively earlier dates
+  const versionMatch = workerdVersion.match(/^1\.(\d{4})(\d{2})(\d{2})/);
+  if (!versionMatch) {
+    return "2024-01-01";
+  }
+  
+  const [, year, month, day] = versionMatch;
+  const versionDateObj = new Date(`${year}-${month}-${day}`);
+  
+  // Try dates going backwards from the version date
+  for (let daysBack = 0; daysBack <= 30; daysBack++) {
+    const testDate = new Date(versionDateObj);
+    testDate.setDate(testDate.getDate() - daysBack);
+    
+    const compatDate = testDate.toISOString().split('T')[0];
+    
+    try {
+      console.log(`  Trying fallback date: ${compatDate}`);
+      const testResult = shell.exec(`node workerd/dump.mjs ${compatDate}`, { silent: true });
+      if (testResult.code === 0) {
+        console.log(`✓ Found working compatibility date: ${compatDate}`);
+        return compatDate;
+      }
+    } catch (error) {
+      console.log(`  Failed: ${error.message}`);
+      continue;
+    }
+  }
+  
+  // Final fallback - use a very early 2024 date
+  return "2024-01-01";
+}
+
+async function calculateSupportPercentage() {
+  // Read the generated workerd data
+  const workerdData = JSON.parse(
+    await fs.readFile(path.join(__dirname, "data", "workerd.json"), "utf8")
+  );
+  
+  const baseline = JSON.parse(
+    await fs.readFile(path.join(__dirname, "data", "baseline.json"), "utf8")
+  );
+  
+  // Use the same logic as generate-table-data.mjs
+  const get = (node, path) => {
+    if (path.length === 0) return node;
+    const [head, ...tail] = path;
+    const value = node[head];
+    if (value == null) return value;
+    return get(value, tail);
+  };
+
+  const isObject = (node) => {
+    return typeof node === "object" && Object.keys(node).length > 0;
+  };
+
+  const isPartOfMockModule = (target, keyPath) => {
+    const moduleName = keyPath[0];
+    const moduleInfo = target[moduleName];
+
+    // mock module can be identified as a module that has only one key - the default,
```

Comment:
This is not correct. Are you sure?

Reply:
I did say vibe coded :)

lemme go through this, am sure not quite right

Reply:
well wait, this is from here

https://github.com/cloudflare/workers-nodejs-compat-matrix/blob/main/generate-table-data.mjs#L70